Step03/app/src/main/java/com/kwoolytech/step03/KwoolyHttp.java

```diff
@@ -0,0 +1,164 @@
+package com.kwoolytech.step03;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+interface KwoolyHttpCallback {
+    void getJsonDataCallback (String result);
+    void getBitmapDataCallback (Bitmap result);
```

Comment:
Maybe name like these more straightforward unless you put the comment ?
onWeatherJsonDataReceived()
onWeatherBitmapDataRecieved()